### PR TITLE
x-pack/filebeat/input/http_endpoint: generalise event path using CEL

### LINKED ISSUE #37910: [Filebeat] http_endpoint - make input compatible with AWS Kinesis http delivery
**Describe the enhancement:**

The [http_endpoint](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-http_endpoint.html) input should be able to accept data in the format given by AWS Kinesis which is an object containing an array of `records`. For example:

```
{
  "requestId": "ed4acda5-034f-9f42-bba1-f29aea6d7d8f",
  "timestamp": 1578090901599
  "records": [
    {
      "data": "aGVsbG8="
    },
    {
      "data": "aGVsbG8gd29ybGQ="
    }
  ]
} 
```

Today the input only accepts objects or arrays.

**Describe a specific use case for the enhancement or feature:**

To receive pushed events from Kinesis.

#### References

- https://aws.amazon.com/blogs/big-data/stream-data-to-an-http-endpoint-with-amazon-kinesis-data-firehose/
- https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

Allow users to handle events where multiple events are provided via single JSON objects in an array field. This is enabled by providing an optional CEL program with only minimal support for CEL evaluations; no extensions from the mito lib extensions are made available.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #37910

### CODE REVIEW

**Review by ShourieG — COMMENTED** [review]
@efd6 Linter errors & Conflict needs to be resolved, otherwise LGTM

**Review by ShourieG — APPROVED** [review]
LGTM

File: x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc [hunk 1]
```diff
@@ -227,6 +227,11 @@ The prefix for the signature. Certain webhooks prefix the HMAC signature with a
 By default the input expects the incoming POST to include a Content-Type of `application/json` to try to enforce the incoming data to be valid JSON.
 In certain scenarios when the source of the request is not able to do that, it can be overwritten with another value or set to null.
 
+[float]
+==== `program`
+
+The normal operation of the input treats the body either as a single event when the body is an object, or as a set of events when the body is an array. If the body should be treated handled differently, for example a set of events in an array field of an object to be handled as a set of events, then a CEL program can be provided through this configuration field. No CEL extensions are provided beyond the function in the CEL standard library. CEL optional types are supported.
```
Comment (kcreddy) [inline x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc:233]: Do you think we should link CEL documentation/standard library for users who may still don't know what CEL means here?

Comment (efd6) [reply x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc]: Added.

**Pull request merged**